includes/functions/functions_lookups.php

```diff
@@ -876,11 +876,11 @@ function zen_run_normal() {
       // show room only
         $zc_run = false;
         break;
-      case (CUSTOMERS_APPROVAL_AUTHORIZATION != '0' and $_SESSION['customer_id'] == ''):
+      case (CUSTOMERS_APPROVAL_AUTHORIZATION != '0' && !zen_is_logged_in()):
       // customer must be logged in to browse
         $zc_run = false;
         break;
-      case (CUSTOMERS_APPROVAL_AUTHORIZATION != '0' and $_SESSION['customers_authorization'] > '0'):
+      case (CUSTOMERS_APPROVAL_AUTHORIZATION != '0' && isset($_SESSION['customers_authorization']) && (int)$_SESSION['customers_authorization'] > 0):
```

Comment:
Should this `isset && (int) > 0` just be `empty()` ?

Reply:
That depends.  The current/previous implementation _kind of_ envisions negative (e.g. -1) values in the 'customers_authorization' session value.  If that's not what's desired, i.e. values will **always** be numeric and greater than or equal to 0, then just using `empty` would suffice.